### PR TITLE
Add K2 analysis switcher 

### PR DESCRIPTION
**Pull request by vmishenev**
It is a part of #3113

### CODE REVIEW

**Review by IgnatBeresnev — COMMENTED** [review]
I don't have much to add to the switch itself, well done here! 

I do have some comments for the test-related changes though. Given that Andrei is waiting for the switch to begin testing, could you please extract all test-related changes into a separate PR (or extract the switch into a separate one)? Then we can quickly merge the switch and unblock Andrei, and then have more time to discuss the tests.

It was also outlined in #3113 for this particular reason, so that we can implement it quickly and unblock testing
> Setting up unit and integration tests to use the switch falls outside of the scope of this issue.

**Review by whyoleg — APPROVED** [review]

**Review by IgnatBeresnev — APPROVED** [review]

File: subprojects/analysis-kotlin-symbols/build.gradle.kts [hunk 1]
```diff
@@ -13,7 +13,7 @@ plugins {
 
 dependencies {
     compileOnly(projects.core)
-    compileOnly(projects.subprojects.analysisKotlinApi)
+    implementation(projects.subprojects.analysisKotlinApi)
```
[the remark this answers is not in the document]
Comment (whyoleg) [reply subprojects/analysis-kotlin-symbols/build.gradle.kts]: This change was not merged in scope of this PR - is it correct? Without it, most likely switch to K2 will not work, no?

[the remark this answers is not in the document]
Comment (IgnatBeresnev) [reply subprojects/analysis-kotlin-symbols/build.gradle.kts]: Only one way to find out :) #3333

[the remark this answers is not in the document]
Comment (vmishenev) [reply subprojects/analysis-kotlin-symbols/build.gradle.kts]: > This change was not merged in scope of this PR - is it correct? Without it, most likely switch to K2 will not work, no?

Sorry, I have missed it( Thank you for noticing it. _It would be nice to have integration tests in single PR._

> (rhetorical question) I wonder how and why it worked before...

Do you mean unit tests? This dependency comes from `pluginBaseTestUtils` where it is not excluded.

**Pull request merged**